sql/src/main/java/io/crate/execution/engine/aggregation/GroupBySingleNumberCollector.java

```diff
@@ -162,13 +198,13 @@ private void iter(Groups groups, Row row) {
             } else {
                 for (int i = 0; i < aggregations.length; i++) {
                     //noinspection unchecked
-                    states[i] = aggregations[i].iterate(ramAccounting,  states[i], inputs[i]);
+                    states[i] = aggregations[i].iterate(ramAccounting, states[i], inputs[i]);
                 }
             }
         }
     }
 
-    private void addWithAccounting(Groups groups, long key, Object[] newStates) {
+    private void addWithAccounting(Groups groups, T key, Object[] newStates) {
```

Comment:
Do we need a different size-constant in `ramAccounting.addBytes` depending on the type?